docs/dapp/README.mdx

```diff
@@ -35,3 +35,60 @@ Sapphire, demonstrates best practices for writing confidential smart contracts
 and working examples for you to explore and extend.
 
 <DocCard item={findSidebarItem('/dapp/opl/')} />
+
+# Consensus Layer RPC Endpoints
+
+:::danger
+
+The RPC endpoint is a *point of trust*. Beside rate limiting, it can also
+perform censorship or even man-in-the-middle attack. If you have security
+considerations, we strongly recommend that you [run your own client
+node][non-validator-node],
+
+:::
+
+[non-validator-node]: ../node/run-your-node/non-validator-node.md
+
+Most dApp developers will build dApps on the ParaTime layer (the *compute*
+layer) mentioned above. However, if you are building a dApp which needs to
+perform consensus operations such as the consensus-layer token transfers,
+ParaTime deposits and withdrawals, governance transactions and similar, you may
+connect to one of the **consensus layer** endpoints (in alphabetic order):
+
+| Provider                    | Mainnet URL          | Testnet URL                  |
+|-----------------------------|----------------------|------------------------------|
+| [Oasis Protocol Foundation] | `grpc.oasis.dev:443` | `testnet.grpc.oasis.dev:443` |
+
+[Oasis Protocol Foundation]: https://oasisprotocol.org
+
+## Consensus Layer Block Explorers
+
+| Provider                    | Mainnet URL                  | Testnet URL                           |
+|-----------------------------|------------------------------|---------------------------------------|
+| [Everstake]                 | https://oasismonitor.com/    | https://testnet.oasismonitor.com/ |
```

Comment:
Let's not include Oasis Monitor until they restore their explorer after the Eden upgrade.